src/deepcode/constants/filesConstants.ts

```diff
@@ -17,3 +17,5 @@ export const FILE_CURRENT_STATUS = {
   same: "same",
   created: "created"
 };
+
+export const SUPPORTED_WATCH_FILES = ['*.jsx', '*.es6', '*.vue', '*.hpp', '*.html', '*.htm', '*.tsx', '*.cxx', '*.es', '*.ts', '*.c', '*.cc', '*.hxx', '*.py', '*.java', '*.h', '*.cpp', '*.js', '.pylintrc', 'tslint.json', '.eslintrc.yml', '.pmdrc.xml', '.ruleset.xml', '.eslintrc.js', '.eslintrc.json', 'ruleset.xml', 'pylintrc'];
```

Comment:
We have an API endpoint to retrieve supported extension and config files (GetFilters). The creation of the watcher may be delayed until you get the filters from the server, or the watched files patterns should be updated once you get the response.

Reply:
done. I refactored watcher in a way that it's activated only once we loaded filters from deepcode server